src/org/traccar/protocol/LaipacProtocolDecoder.java

```diff
@@ -88,28 +101,86 @@ protected Object decode(
         dateBuilder.setDateReverse(parser.nextInt(0), parser.nextInt(0), parser.nextInt(0));
         position.setTime(dateBuilder.getDate());
 
-        String type = parser.next();
+        String eventCode = parser.next();
+        String decodedAlarm = decodeAlarm(eventCode);
+        if (decodedAlarm != null) {
+            position.set(Position.KEY_ALARM, decodeAlarm(eventCode));
```

Comment:
Why do you call `decodeAlarm` twice? You don't need null check here and probably don't need variable as well.